src/voodoo/conversations/VoodooConversation.ts

```diff
@@ -65,30 +77,62 @@ export default class VoodooConversation {
     // Wait for all promises in decryptedMessages to resolve
     const results = await Promise.all(decryptedMessages)
 
+    let messages = this.multiSession.messages.get(sessionId)
+    if (!messages) {
+      messages = []
+    }
+
     for (const m of results) {
       if (m) {
-        this._messages.push(m)
+        messages.push(m)
       }
     }
     // Sort all messages
-    this._messages.sort((a: VoodooMessage, b: VoodooMessage) => {
+    messages.sort((a: VoodooMessage, b: VoodooMessage) => {
       return a.timestamp - b.timestamp
     })
-    return this._messages
+    this.multiSession.messages.set(sessionId, messages)
+    return messages
+  }
+
+  // Provides an aggregated and sorted list of messages for all of the device sessions
+  async messages(): Promise<VoodooMessage[]> {
```

Comment:
Having messages spread across multiple topics is going to make pagination much more annoying in the short term (given that we can't do it in a single query anymore). Once we have PDS, it probably matters less given that our goal is always to fetch everything new and put it into PDS.

Reply:
Agreed - it's something we'll want to experiment with and learn about in this prototyping phase!